### PR TITLE
Fixed empty tabs showing to non admin users

### PR DESCRIPTION
**Pull request by PaarthAgarwal**
Fix for #8593.

### CODE REVIEW

File: wagtail/admin/tests/test_edit_handlers.py [hunk 1]
```diff
@@ -460,6 +460,45 @@ def test_render_form_content(self):
...
+        )
+        result = tabbed_interface.render_html()
+        self.assertNotIn(
+            '<a id="tab-label-event-details" href="#tab-event-details" class="w-tabs__tab shiny" role="tab" aria-selected="false" tabindex="-1">',
+            result,
+        )
+        self.assertIn(
+            '<a id="tab-label-speakers" href="#tab-speakers" class="w-tabs__tab " role="tab" aria-selected="false" tabindex="-1">',
+            result,
+        )
```
[the remark this answers is not in the document]
Comment (PaarthAgarwal) [reply wagtail/admin/tests/test_edit_handlers.py:500]: Sure

File: wagtail/admin/tests/test_edit_handlers.py [hunk 2]
```diff
@@ -460,6 +460,45 @@ def test_render_form_content(self):
         # in the panel definition
         self.assertNotIn("signup_link", result)
```
[the remark this answers is not in the document]
Comment (lb-) [reply wagtail/admin/tests/test_edit_handlers.py]: Ok. I'll raise a seperate bug for the InlinePanel thing probably - pretty sure that should support the argument but I'll check.

**Comment by lb-** [comment]
@PaarthAgarwal please go ahead and add panels to the `ObjectList([], heading="Dinosaurs"),` in the test data so that it has something that shows. even just a `HelpPanel`.

We will assume that empty tabs should never show and this setup in the unit test was incorrect.

File: wagtail/admin/tests/test_edit_handlers.py [hunk 3]
```diff
@@ -460,6 +460,46 @@ def test_render_form_content(self):
...
 
+    def test_tabs_permissions(self):
+        # self.assertIn("Abergavenny sheepdog trials</textarea>", "result")
+        # test that two tabs show when the current has permission to see both
+        # check that one tab shows when the current user has permission to only single one
+        event_page_tabbed_interface = TabbedInterface(
```
Comment (lb-) [inline wagtail/admin/tests/test_edit_handlers.py:467]: now that you can use the `self.event_page_tabbed_interface` you should not need to build a custom one here.

**Comment by lb-** [comment]
@PaarthAgarwal  looking good, once you update the dinosaurs tabs (something like as follows).

```
# Test overriding edit_handler with a custom one
StandardChild.edit_handler = TabbedInterface(
    [
        ObjectList(StandardChild.content_panels, heading="Content"),
        ObjectList(StandardChild.promote_panels, heading="Promote"),
        ObjectList(
            StandardChild.settings_panels, heading="Settings", classname="settings"
        ),
        ObjectList([HelpPanel('ARRR')], heading="Dinosaurs"), ## CHANGED
    ],
    base_form_class=WagtailAdminPageForm,
)
```

After this you can run the subset to tests locally that are failing right now

`python runtests.py wagtail.admin.tests.pages.test_create_page`

If they pass - you should be good.

Remember to run `make format` before pushing those final changes up to your remote branch.

Hopefully all the tests will pass and we can lock this in, remember to mark your PR as a non-draft when ready.

**Review by lb- — APPROVED** [review]
Great work here! thanks for working through adding unit tests and learning how to do this @PaarthAgarwal

**Comment by PaarthAgarwal** [comment]
> Great work here! thanks for working through adding unit tests and learning how to do this @PaarthAgarwal

Thanks sensei✌🏼

**Pull request merged**